### PR TITLE
Feature/interactive_physical_layout

### PR DESCRIPTION
**Pull request by jbrekelmans**
1/3 Pull requests on repositories Netflix/vizceral, Netflix/vizceral-example and Netflix/vizceral-react with the following features around "optimizing layout manually":
1. Repositioning of nodes by dragging them
2. Interactive Particle System simulation (nodes are particles and edges are spring forces) because it's awesome and allow for good automatic layout on some graphs
3. Resetting layout if #2 get's out of hand
4. Layout algorithm that respects positional metadata in the JSON graph files and otherwise lays out nodes in a circle
5. Export (well, Console.log) current node positions to be used as positional metadata once a satisfactory manual layout has been created

How to run:
1. Clone the branch feature/interactive_physical_layout of repositories jbrekelmans/vizceral, jbrekelmans/vizceral-example and jbrekelmans/vizceral-react locally
2. Run command 'npm link' in vizceral
3. Run command 'npm link vizceral' in vizceral-react
4. Run command 'npm link' in vizceral-react
5. Run command 'npm link vizceral-react' in vizceral-example
6. Make sure a graph (e.g. in sample_data.json) with 10-20 nodes is loaded for best results using the default Particle System settings

### CODE REVIEW

File: src/vizceral.jsx [hunk 1]
```diff
@@ -200,18 +217,18 @@ Vizceral.defaultProps = {
   connectionHighlighted: () => {},
   definitions: {},
   filters: [],
-  graphsUpdated: () => {},
+  graphsUpdated: () => undefined,
```
Comment (jrsquared) [inline src/vizceral.jsx]: Why the change? These have default empty functions so they are just no-ops if the user does not provide listeners to things they do not care about.

**Comment by jbrekelmans** [comment]
Hey Justin!

I've changed the condition for calling updateData so it works for me. Of course I would like to know whether it works for other people as well and what the design here is.

The main difference is I call updateData if regions are removed and I do not accept non-falsy 'nodes' that are not arrays. 
What you'll additionally see is my defensive style of coding preferring verbosity/confidence/debuggability over having less lines of code and performance (but performance-improvements are not really significant in this code path).  

Kind regards,

Jasper Brekelmans
